src/operator/controllers/intents_reconcilers/otterize_cloud/cloud_reconciler.go

```diff
@@ -0,0 +1,98 @@
+package otterize_cloud
+
+import (
+	"context"
+	"fmt"
+	"github.com/Khan/genqlient/graphql"
+	otterizev1alpha1 "github.com/otterize/intents-operator/src/operator/api/v1alpha1"
+	"github.com/otterize/intents-operator/src/operator/controllers/intents_reconcilers/otterize_cloud/graphql_clients/intents"
+	"github.com/otterize/intents-operator/src/shared/injectablerecorder"
+	"github.com/samber/lo"
+	"github.com/sirupsen/logrus"
+	"golang.org/x/oauth2"
+	"golang.org/x/oauth2/clientcredentials"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+type OtterizeCloudReconciler struct {
+	client.Client
+	Scheme    *runtime.Scheme
+	tokenSrc  oauth2.TokenSource
+	cloudAddr string
+	injectablerecorder.InjectableRecorder
+}
+
+func NewOtterizeCloudReconciler(
+	client client.Client,
+	scheme *runtime.Scheme,
+	otterizeClientID,
+	otterizeClientSecret,
+	cloudAddr string) *OtterizeCloudReconciler {
+
+	cfg := clientcredentials.Config{
```

Comment:
This bit of code is copypasted here and in NamespaceWatcher - factor it out to a function that returns `graphql.Client` and use that instead.

A nice trick you can do is add `viper` configuration, bind it to pflags, and then have that function use the client ID, client secret and cloud addr from the configuration directly, rather than passing it around (since they are global - the same for the operator no matter where the code is placed). This way the function will only need to take the URI subpath, such as `intents/query`.

Check out `GetAPITokenSource` in the old CLI, and the `viper` docs: https://pkg.go.dev/github.com/spf13/viper#readme-working-with-flags

Reply:
Will check it out, but see comment above - will probably get removed from the watcher anyway.